tools/swift-format/Sources/Rules/BlankLineBetweenMembers.swift

```diff
@@ -17,5 +17,148 @@ import SwiftSyntax
+                                      leadingTrivia: correctTrivia) as! MemberDeclListItemSyntax
+        hasValidNumOfBlankLines = false
+        membersList.append(newMember)
+      }
+      // Ensures that there is at least one blank line between each member of a type.
+      // Unless is a single-line declaration and the format is configured to
+      // ignored them.
+      else if case .newlines(let numNewLines) = firstPiece,
+              !ignoreItem(item: currentMember),
+              numNewLines - 1 == 0 {
+        let numBlankLines = member != node.members.first ? 1 : 0
+        let correctTrivia = Trivia.newlines(numBlankLines) + memberTrivia
+        let newMember = replaceTrivia(on: currentMember, token: currentMember.firstToken!,
+                                      leadingTrivia: correctTrivia) as! MemberDeclListItemSyntax
+        
+        diagnose(.addBlankLine, on: currentMember)
+        hasValidNumOfBlankLines = false
+        membersList.append(newMember)
+      }
+      else {
+        membersList.append(member)
+      }
+    }
+    
+    return hasValidNumOfBlankLines ? node :
+           node.withMembers(SyntaxFactory.makeMemberDeclList(membersList))
+  }
+  
+  /// Indicates if the given trivia has more than
+  /// the maximum number of blank lines.
+  func exceedsMaxBlankLines(_ trivia: Trivia) -> Bool {
+    let maxBlankLines = context.configuration.maximumBlankLines
 
+    for piece in trivia {
+      if case .newlines(let num) = piece,
+        num - 1 > maxBlankLines {
+        return true
+      }
+    }
+    return false
+  }
+  
+  /// Returns the given trivia without any set of consecutive blank lines
+  /// that exceeds the maximumBlankLines.
+  func removeExtraBlankLines(_ trivia: Trivia, _ member: MemberDeclListItemSyntax) -> Trivia {
+    let maxBlankLines = context.configuration.maximumBlankLines
+    var pieces = [TriviaPiece]()
+    
+    // Iterates through the trivia, verifying that the number of blank
+    // lines in the file do not exceed the maximumBlankLines. If it does
+    // a lint error is raised.
+    for piece in trivia {
+      if case .newlines(let num) = piece,
+         num - 1 > maxBlankLines {
+        pieces.append(.newlines(maxBlankLines + 1))
+        diagnose(.removeBlankLines(count: num - maxBlankLines)
+                 ,on: member)
+      }
+      else {
+        pieces.append(piece)
+      }
+    }
+    return Trivia.init(pieces: pieces)
+  }
+  
+  /// Indicates if a declaration has to be ignored by checking if it's
+  /// a single line and if the format is configured to ignore single lines.
+  func ignoreItem(item: MemberDeclListItemSyntax) -> Bool {
+    guard let firstToken = item.firstToken else { return false }
+    guard let lastToken = item.lastToken else { return false }
+    
+    let isSingleLine = firstToken.positionAfterSkippingLeadingTrivia.line ==
+                       lastToken.positionAfterSkippingLeadingTrivia.line
```

Comment:
Our wrapping rules say that this should only be indented +2 from the previous line, not aligned after the `=`. (Likewise just below.)